### PR TITLE
Revise and refactor Prometheus metrics.

### LINKED ISSUE #553: Metric naming
Hi,

This is some late cycle feedback. I did not spot this during the PR - oops. But we were just working on alerts for the new `routinator:unstable` version.

I noticed this set of metrics:
```
routinator_repo_rejected_points{uri="https://$repo"} 0
routinator_repo_invalid_manifests{uri="https://$repo"} 0
routinator_repo_stale_manifests{uri="https://$repo"} 0
routinator_repo_missing_manifests{uri="https://$repo"} 0
routinator_repo_invalid_crls{uri="https://$repo"} 0
routinator_repo_stale_crls{uri="https://$repo"} 0
routinator_repo_stray_crls{uri="https://$repo"} 0
routinator_repo_valid_ee_certs{uri="https://$repo"} 0
routinator_repo_invalid_certs{uri="https://$repo"} 0
routinator_repo_invalid_roas{uri="https://$repo"} 0
routinator_repo_valid_gbrs{uri="https://$repo"} 0
routinator_repo_invalid_gbrs{uri="https://$repo"} 0
routinator_repo_other_objects{uri="https://$repo"} 0
routinator_repo_vrps_unsafe{uri="https://$repo"} 0
routinator_repo_vrps_filtered_locally{uri="https://$repo"} 0
```

It may be more idiomatic (with regards to the alerts that you rewrite) if these restructured to match this pattern:
```
routinator_repo_object{type="vrps", state="filtered_locally"} 0
```

With the current metrics our alert ends up looking like:
```
(routinator_repo_rejected_points{uri="$repo"} > 0) 
or
(routinator_repo_invalid_manifests{uri="$repo"} > 0)
or
(routinator_repo_stale_manifests{uri="$repo"} > 0)
...
```

I would prefer an alerting query like
```
routinator_repo_object{uri="$repo",state=~".*(rejected|invalid|stale).*"}
```
If possible.

Whether this refactoring is still possible depends on how many alerts already exist…

### PR DESCRIPTION
**Pull request by partim**
This PR revises the new Prometheus metrics in an attempt to make them more canonical.

Resolves #553.

### CODE REVIEW

**Comment by AlexanderBand** [comment]
```
overlapping with rejected publication points  per TAL
```
has a double space between `points` and `per`

File: src/http/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,802 @@
...
+            .value(metrics.duplicate);
+        target.multi(contributed_metric).label(group.label(), name)
+            .value(metrics.contributed);
+    }
+}
+
+fn rrdp_metrics(target: &mut Target, metrics: &[RrdpRepositoryMetrics]) {
+    let status = Metric::new(
+        "rrdp_status",
+        "combines status code for RRDP update requests",
```
Comment (DRiKE) [inline src/http/metrics.rs:304]:
```suggestion
        "combined status code for RRDP update requests",
```

File: src/http/metrics.rs [hunk 2]
```diff
@@ -0,0 +1,802 @@
...
+            };
+        }).for_each(|(addr, update)| {
+            match update {
+                Some(update) => {
+                    let duration = Utc::now() - update;
+                    target.multi(item).label("addr", addr).value(
+                        format_args!(
+                            "{}.{:03}",
+                            duration.num_seconds(),
+                            duration.num_milliseconds() % 1000,
```
Comment (DRiKE) [inline src/http/metrics.rs:478]: `subsec_millis` for consistency?

File: src/http/metrics.rs [hunk 3]
```diff
@@ -0,0 +1,802 @@
...
+            "http_current_connections",
+            "number of currently open HTTP connections",
+            MetricType::Gauge
+        ),
+        metrics.conn_open() - metrics.conn_close()
+    );
+    target.single(
+        Metric::new(
+            "http_bytes_read",
+            "number of bytes read from HTTP connection",
```
Comment (DRiKE) [inline src/http/metrics.rs:534]:
```suggestion
            "number of bytes read from HTTP connections",
```

File: src/http/metrics.rs [hunk 4]
```diff
@@ -0,0 +1,802 @@
...
+            MetricType::Counter
+        ),
+        metrics.bytes_read()
+    );
+    target.single(
+        Metric::new(
+            "http_bytes_written",
+            "number of bytes written ot HTTP connections",
```
Comment (DRiKE) [inline src/http/metrics.rs:542]:
```suggestion
            "number of bytes written to HTTP connections",
```

**Review by DRiKE — APPROVED** [review]

File: src/http/metrics.rs [hunk 5]
```diff
@@ -0,0 +1,802 @@
...
+            .value(metrics.others);
+    }
+}
+
+fn vrp_metrics<'a>(
+    target: &mut Target, group: Group,
+    metrics: impl Iterator<Item = (&'a str, &'a VrpMetrics)>
+) {
+    let valid_metric = Metric::with_prefix(
+        group.prefix(), "valid_vrps_total",
```
Comment (AlexanderBand) [inline src/http/metrics.rs:257]: Incomplete help text, e.g.
```
# HELP routinator_ta_valid_vrps_total per trust anchor
# HELP routinator_repository_valid_vrps_total per repository

File: src/http/metrics.rs [hunk 6]
```diff
@@ -0,0 +1,802 @@
...
+            "rtr_bytes_read",
+            "total number of bytes read from RTR connections",
+            MetricType::Counter
+        ),
+        metrics.bytes_read()
+    );
+    target.single(
+        Metric::new(
+            "rtr_bytes_written",
+            "total number of bytes weitten to RTR connections",
```
Comment (AlexanderBand) [inline src/http/metrics.rs:412]: s/weitten/written

**Review by AlexanderBand — CHANGES_REQUESTED** [review]

File: src/http/metrics.rs (hunk 2 above)
Comment (partim) [reply src/http/metrics.rs:478]: This is actually a `chrono::Duration` that doesn’t have that method.

**Review by AlexanderBand — APPROVED** [review]

**Pull request merged**